theano/tensor/raw_random.py

```diff
@@ -520,13 +520,8 @@ def binomial(random_state, size=None, n=1, p=0.5, ndim=None,
     n = tensor.as_tensor_variable(n)
     p = tensor.as_tensor_variable(p)
     ndim, size, bcast = _infer_ndim_bcast(ndim, size, n, p)
-    if n.dtype == 'int64':
-        # THIS WORKS AROUND A NUMPY BUG on 32bit machine
-        # Erase when the following works on a 32bit machine:
```

Comment:
I would put that code in the test. In fact, depending of the consequence of the bug (error vs segfault) we could detect it here and apply the work around only if we need it. I just check it with numpy 1.8.0 and this raise an error. So I would change all this to:


```
if n.dtype == 'int64':
   try:
       numpy.random.binomial(n=numpy.asarray([2,3,4], dtype='int64'), p=numpy.asarray([.1, .2, .3], dtype='float64'))
   except TypeError:
      n = tensor.cast(n, 'int32')
```
So we will apply the fix only when needed. So we won't risk a silent bug with python 64 bit.

Something even better would be to modify RandomFunction to include a hack to check if all the int64 values fit in a int32 values, but this is probably too much.

What do you think?